src/gui/editors/SongEditor.cpp

```diff
@@ -53,19 +53,40 @@
 #include "Track.h"
 
 positionLine::positionLine( QWidget * parent ) :
-	QWidget( parent )
+	QWidget( parent ),
+	m_tailGradient ( false ),
+	m_lineColor (0, 0, 0, 0)
 {
-	setFixedWidth( 1 );
+	setFixedWidth( 8 );
 	setAttribute( Qt::WA_NoSystemBackground, true );
+	setAttribute( Qt::WA_TransparentForMouseEvents );
 }
 
-
-
-
 void positionLine::paintEvent( QPaintEvent * pe )
 {
 	QPainter p( this );
-	p.fillRect( rect(), QColor( 255, 255, 255, 153 ) );
+	
+	// Create the gradient trail behind the line
+	QLinearGradient gradient(rect().bottomLeft(), rect().bottomRight());
+	
+	gradient.setColorAt(0,
+		QColor (m_lineColor.red(), m_lineColor.green(), m_lineColor.blue(), 0) );
+	gradient.setColorAt(1,
+		QColor (m_lineColor.red(), m_lineColor.green(), m_lineColor.blue(), 153) );
+		
+	if (m_tailGradient) 
+	{ 
+		gradient.setColorAt(0.875, 
+			QColor (m_lineColor.red(), m_lineColor.green(), m_lineColor.blue(), 60) );
+	}
+	else 
+	{ 
+		gradient.setColorAt(0.875, 
+			QColor (m_lineColor.red(), m_lineColor.green(), m_lineColor.blue(), 0) );
+	}
```

Comment:
Again, a total nitpick, but might I suggest something like:

```cpp
	int opacity;
	if (m_tailGradient) 
	{ 
		opacity = 60;
	}
	else 
	{ 
		opacity = 0;
	}
	gradient.setColorAt(0.875, 
		QColor (m_lineColor.red(), m_lineColor.green(), m_lineColor.blue(), opacity) );
```

Reply:
I would favour a ternary expression here; it's always nice if you can initialise variables at point of declaration, and mark them const too. (And it uses far less vertical space too.)
```cpp
const auto opacity = m_tailGradient ? 60 : 0;
```

Alternatively, to avoid manually copying the red, green, and blue components, which might be a bit easier to read:
```cpp
auto lineColor = m_lineColor;
lineColor.setAlpha(m_tailGradient ? 60 : 0);
gradient.setColorAt(0.875, lineColor);
```

Reply:
Oh of course, that didn't even cross my mind. Good suggestion!

Reply:
> to avoid manually copying the red, green, and blue components, which might be a bit easier to read

Something I was looking to do as well. I'll implement the last one, that seems the most straightforward and short solution. Thanks for the suggestion!